synapse/handlers/oidc_handler.py

```diff
@@ -1071,3 +1108,9 @@ async def map_user_attributes(
                 display_name = None
 
         return UserAttribute(localpart=localpart, display_name=display_name)
+
+    async def get_extra_attributes(self, userinfo: UserInfo, token: Token) -> JsonDict:
+        extras = {}  # type: Dict[str, str]
+        for key, template in self._config.extra_attributes.items():
+            extras[key] = template.render(user=userinfo).strip()
```

Comment:
wonder if this could do with some exception handling so that we know which attribute is failing to render.

Reply:
I kind of just followed the above code which also does not do exception handling. It could probably be improved though. 😄

Reply:
The ones above don't catch since the handler catches any exception and converts it to a mapping error. I think I'll just log an error in this case and skip that key?